tests/integration/components/args-decorator-test.ts

```diff
@@ -21,6 +21,18 @@ module('Integration | Component | arg-decorator', function(hooks) {
     assert.dom('.level').hasNoText('@arg will pass `undefined` as a default when an initializer is not present');
   });
 
+  test('it uses the correct context in property initializers', async function(assert) {
+    await render(hbs`
+      <Character @name="link" />
+      <Character @name="zelda" />
+    `);
+
+    const linkId = this.element.querySelector('.character:first-child .id');
+    const zeldaId = this.element.querySelector('.character:last-child .id');
+
+    assert.notEqual(linkId.innerText, zeldaId.innerText);
+  });
```

Comment:
Hey @bertdeblock, thanks for reporting this bug! If you would like to contribute a test for this issue, can we do something like this in addition to your two component test:

```js
  skip('it uses the correct context in property initializers', async function(assert) {
    await render(hbs` <Character @name="link" /> `);
    const argId = this.element.querySelector('.id')!.textContent;
    const privateId  = this.element.querySelector('._id')!.textContent;
    assert.equal(argId, privateId, '@arg calls the default getter with the correct context');
  });
```
where `_id` is a non-`@arg` getter property that returns `guidFor(this)`

Let's mark the test as `skip` so the build passes. I can follow up with a fix shortly. Let me know what you would like to do.

Reply:
@jkusa Should be good now! I removed my initial test if that's okay. It felt redundant when adding your suggestion.